kyo-prelude/shared/src/main/scala/kyo2/Abort.scala

```diff
@@ -0,0 +1,93 @@
+package kyo2
+
+import kernel.<
+import kernel.Const
+import kernel.Effect
+import kernel.Frame
+import kyo.Tag
+import scala.reflect.ClassTag
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
+
+sealed trait Abort[-E] extends Effect[Const[E], Const[Unit]]
+
+object Abort:
+
+    private inline def erasedTag[E]: Tag[Abort[E]] = Tag[Abort[Any]].asInstanceOf[Tag[Abort[E]]]
+
+    inline def fail[E](inline value: E): Nothing < Abort[E] =
+        Effect.suspend[Any](erasedTag[E], value, _ => ???)
+
+    inline def when[E](b: Boolean)(inline value: E)(
+        using inline tag: Tag[Abort[E]]
+    ): Unit < Abort[E] =
+        if b then fail(value)
+        else ()
+
+    class GetOps[E >: Nothing](ign: Unit) extends AnyVal:
+        inline def apply[A](either: Either[E, A]): A < Abort[E] =
+            either match
+                case Right(value) => value
+                case Left(value)  => fail(value)
+
+        inline def apply[A](opt: Option[A]): A < Abort[None.type] =
+            opt match
+                case None    => fail(None)
+                case Some(v) => v
+
+        inline def apply[T](e: Try[T]): T < Abort[Throwable] =
+            e match
+                case Success(t) => t
+                case Failure(v) => fail(v)
+    end GetOps
+
+    inline def get[E >: Nothing]: GetOps[E] = GetOps(())
+
+    class RunOps[E >: Nothing](ign: Unit) extends AnyVal:
+        def apply[E0 <: E, A, B, ES, ER](v: A < (Abort[ES] & B))(
+            using
+            h: HasAbort[E0, ES] { type Remainder = ER },
+            ct: ClassTag[E],
+            frame: Frame
+        ): Either[E, A] < (ER & B) =
+            Effect.catching {
+                Effect.handle(erasedTag[E], v.map(Right(_): Either[E, A]))(
+                    accept = [C] =>
+                        (input, _) =>
+                            (input.asInstanceOf[Any]) match
+                                case input: E => true
+                                case _        => false,
+                    handle = [C] => (input, _) => Left(input)
+                ).asInstanceOf[Either[E, A] < ER & B]
+            } {
+                case fail: E => Left(fail)
+            }
+    end RunOps
+
+    inline def run[E >: Nothing]: RunOps[E] = RunOps(())
```

Comment:
Improvements of the `*Dsl` classes:

1. Renamed to `*Ops`, I think `Dsl` can be confusing to users
2. The `Ops` classes are now `AnyVal` so the compiler can avoid emitting code for their allocation
3. Inlined the methods that produce `Ops` classes

We should migrate to [clause interleaving](https://docs.scala-lang.org/sips/clause-interleaving.html) once it's not experimental anymore.

Reply:
> Can you change name for Unit from ign -> dummy?

Good point! I'm renaming it to `dummy`.

> Additionally, is there a benefit to inlining the def run?

It saves a few bytecodes because calling a method uses more instructions than referring to `()`. It's possible to see that in `BytecodeTest`. If I remove the `inline` from `Effect.suspend`, the test fails with:

```
[info] BytecodeTest:
[info] - suspend *** FAILED *** (34 milliseconds)
[info]   Map("test" -> 27) did not equal Map("test" -> 20) (BytecodeTest.scala:25)
[info]   Analysis:
[info]   Map("test": 27 -> 20)
[info] - suspendMap (14 milliseconds)
[info] - map (8 milliseconds)
[info] - handle (6 milliseconds)
```